squidDrone/src/com/com_message_buffer.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef SRC_COM_MESSAGE_BUFFER_HPP_
+#define SRC_COM_MESSAGE_BUFFER_HPP_
+
+#include <array>
+#include <queue>
+#include "com_types.hpp"
+
+namespace com {
+  /** 
+    * @brief This class represents a queue style buffer for communication
+    * frames exchanged between drones.
+    * 
+    */
+  class ComMessageBuffer {
+    public:
+      /** 
+       * @brief Construct a new Com Message Buffer object
+       * 
+       */
+      ComMessageBuffer(){};
+
+      /** 
+       * @brief Transfer data to queue buffer. 
+       * 
+       * @param data Reference to data frame. Must conform to data frame specifications.
```

Comment:
You say the data frame mus conform to data frame specifications, but never test it in your implementation

Reply:
This concerns the fixed array length. I tested it and found out, that an array with a different length is not accepted at compile time. Data frame length is part of the data frame specifications.

Reply:
Ah ok. I see - resolved